### ISSUE TITLE
extra-1.6.4 test suite failure

### ISSUE THREAD

**Issue by peti** [ndmitchell/extra#33]
This happened with GHC 8.4.1:

~~~
Test suite extra-test: RUNNING...
\(x :: IO Int) -> void (once x) == return ()
*** Failed! Falsifiable (after 4 tests): 
<<IO>>
Test suite extra-test: FAIL
~~~

The error occurs only spuriously, though.

**Comment by ndmitchell** [comment]
Thanks for the info. I've been running on the Linux Travis CI for a few months without observing any failure there. Which OS?

**Comment by peti** [comment]
NixOS Linux x86_64.

**Comment by ndmitchell** [comment]
Were you running it successfully with 8.2 for a long time? How often are you observing failures? (e.g. 10 times out of 100, or just once or twice). It's the first test, which makes me wonder if it's not the test itself but mechanisms behind the test.

**Comment by peti** [comment]
Yes, the error never occurred with 8.2.x in an lts-10.x package set. I saw it with 8.4.1 once so far. I haven't been running many build with that compiler yet, obviously, since it was release only yesterday. So we don't have a large sample size.

**Comment by ndmitchell** [comment]
Please let me know if it repeats - once can always be quantum effects but twice would be some underlying issue somewhere...

**Comment by ndmitchell** [comment]
I have improved the error message in case such a test failure happens again, and released 1.6.5 with that improved test suite.

**Comment by ndmitchell** [comment]
Closing until another instance crops up.
